### DISCUSSION TITLE
[1주차] 2024-10-16 Scrum Meeting 일지

### DISCUSSION THREAD

**Discussion by yeon-so** [nhnacademy-be7-nuri-books/nuri-books-demo#4] [회의록]
<details>
<summary>목차</summary>
<div markdown="1">

- [시작 전 확인 사항 (과제)](#---------------)
- [회의 진행 내역](#--------)
  * [ERD](#erd)
  * [기타](#--)
- [내일 할일](#-----)

</div>
</details>

---

# 시작 전 확인 사항 (과제)
> [!NOTE]
> ~~1. 프로젝트에 적용할 뷰 템플릿 찾아오기~~ :EMOJICODE:
> 2. 포장지 및 DB 설계에 대하여 생각해보기

---

# 회의 진행 내역
## ERD

> [!NOTE]
> - __추후 확인해야 할 안건__
> - [x] 포장 확장 기능 구현/미구현 여부 생각하기 
> - [ ] 주문 관련 회의는 결제 떄 일괄 회의
> - [x] 쿠폰의 기간 분리를 해야하는 가. (쿠폰은 전체적으로 문제.) - 우선은 분리 안함 (로직 처리의 편의성을 위해)
> - [x] 사용되는 모든 이미지를 따로 테이블로 분리하여 관리할 것인가?  - 분리

- 도서
  - 도서 종류 별로 포장 일괄 작업
    - (2024-10-16) 현재는 아래 항목 무시)
  - __필요 시)__ 단 한 종류의 도서를 여러 권 구입 시 확장
    - (n권 당 가격 추가)

- 주문
  - 일부 반품이 발생하면 해당 주문은 취소 처리하고 반품이 일어나지 않은 물품으로 새로운 주문을 생성한다. 
  - 주문 상세의 개별 가격은 사용자가 결제한 금액을 말한다.
 
- 장바구니
  - 비회원 : Session Redis 에 저장
  - 회원 : 로그인 시 Session Redis에 저장, 로그아웃 시 해당 정보를 DB로 Insert

## 기타
- 모든 상태값은 테이블 분리로 관리

> [!TIP]
> 1. status 현재 상태를 나타내고, state는 변하는 상태를 나타낸다.
>   - 즉, 상태 테이블은 state로 쓰는 게 맞다.
> 2. DB의 테이블을 따로 뺄 시 확장성을 고려하자. 
>   - 확장을 할 필요가 없는 값이라면 Enum으로
---

# 내일 할일
- [ ] 내일 2시 : 1차 ERD 리뷰
- [ ] 2시 안에 컬럼명, 데이터타입 정리
- [ ] 9시 안에 장바구니 - 주문 - 결제 시나리오 생각해오기

**Comment by yongtaek2** [comment]
누리는 내가 아는 ~~프론트엔드~~ 백엔드 엔지니어 중 제일이다.
